receptorctl/tests/test_cli.py

```diff
@@ -68,6 +101,40 @@ def test_cmd_work_list_successfully(self, invoke):
         # Require fixture with a node running work
         pass
 
+    def test_cmd_work_results_invalid_unit_id(self, invoke):
+        """Test results command with an invalid work unit ID"""
+        result = invoke(commands.work, ["results", "invalid-unit-id"])
+        assert result.exit_code != 0
+        assert result.exception is not None
+        assert "invalid-unit-id" in str(result.exception)
+
+    def test_cmd_work_results_successful(self, invoke, default_receptor_controller_socket_file):
+        node1_controller = default_receptor_controller_socket_file
+
+        # Submit a simple echo work unit
+        payload = "test-output-data"
+        work = node1_controller.submit_work("echo-uppercase", payload, node="node3")
+        unit_id = work.pop("unitid")
+
+        # Wait for work to complete
+        timeout_seconds = 10
+        work_completed = False
+        for _ in range(timeout_seconds):
```

Comment:
Minor nit. This variable could be renamed to `max_retries`

```suggestion
        # Wait for work to complete
        max_retries = 10
        work_completed = False
        for _ in range(max_retries):
```

Reply:
Done :pray: